algebras/algebra/src/main/scala/endpoints/algebra/Urls.scala

```diff
@@ -73,6 +75,11 @@ trait Urls {
     */
   type QueryStringParam[A]
 
+  def refineQueryStringParam[A, B](pa: QueryStringParam[A])(f: A => Option[B])(g: B => A): QueryStringParam[B]
+
+  /** Ability to define `UUID` query string parameters */
+  implicit def uuidQueryString: QueryStringParam[UUID]
```

Comment:
You should now be able to implement `uuidQueryString` here, instead of reimplementing it in each interpreter ;)

Reply:
Right, I removed most of them except for thost in openapi.Urls and uuidSegment in server.Urls.